installer/installer.go

```diff
@@ -27,27 +26,82 @@ import (
 	"github.com/pkg/errors"
 )
 
-type UInstaller interface {
-	InstallUpdate(io.ReadCloser, int64) error
-	EnableUpdatedPartition() error
+type Rebooter interface {
+	Reboot() error
 }
 
-func Install(art io.ReadCloser, dt string, key []byte, scrDir string,
-	device UInstaller, acceptStateScripts bool) error {
+type PayloadInstaller interface {
```

Comment:
> Does not feel like PayloadInstaller. More like UpdatePerformer or something.

Ok. I would like "payload" to be in the name though, since it maps to our taxonomy for artifacts. A lot of places in the source code when we say "update" we really mean "payload", and the reason for this is mostly historical (it would be a lot of effort to change it everywhere). We should use it for new code though, so I suggest `PayloadUpdatePerformer`.

> Also maybe grouping into more granular interfaces could make sense instead of creating one. Logically I would put all the actionable methods inside one and all the info getting into another one.

I get your point, but I'm wondering how much sense it makes in this particular case. This interface is more or less a mapping to our [update module protocol](https://github.com/mendersoftware/mender/blob/update_modules/Documentation/update-modules-v3-file-api.md#states-and-execution-flow), so it seems pretty unlikely that we will use any of these interfaces in isolation. For example calling `NeedsReboot` has strict ordering requirements according to the protocol, so you can't just call it whenever you want, you have to call it after `InstallUpdate`. So I think it makes sense to keep them together.